packages/api/lambdas/sqs-message-consumer.js

```diff
@@ -18,66 +18,95 @@ const Rule = require('../models/rules');
  * messages from SQS queue are processed
  */
 async function processQueues(event, dispatchFn) {
-  const model = new Rule();
-  const rules = await model.getRulesByTypeAndState('sqs', 'ENABLED');
+  const rulesModel = new Rule();
+  const rules = await rulesModel.queryRules({
+    type: 'sqs',
+    state: 'ENABLED'
+  });
 
   const messageLimit = event.messageLimit || 1;
   const timeLimit = event.timeLimit || 240;
 
-  await Promise.all(rules.map(async (rule) => {
+  const rulesByQueueMap = rules.reduce((map, rule) => {
     const queueUrl = rule.rule.value;
+    // eslint-disable-next-line no-param-reassign
+    map[queueUrl] = map[queueUrl] || [];
+    map[queueUrl].push(rule);
+    return map;
+  }, {});
+
+  await Promise.all(Object.keys(rulesByQueueMap).map(async (queueUrl) => {
+    const rulesForQueue = rulesByQueueMap[queueUrl];
 
-    if (!(await sqsQueueExists(queueUrl))) return Promise.resolve();
+    if (!(await sqsQueueExists(queueUrl))) {
+      const ruleIds = rulesForQueue.map((rule) => rule.id);
+      log.info(`Could not find queue ${queueUrl}. Unable to process rules ${ruleIds}`);
+      return Promise.resolve();
+    }
+
+    // Use the max of the visibility timeouts for all the rules
+    // bound to this queue.
+    const visibilityTimeout = rulesHelpers.getMaxTimeoutForRules(rulesForQueue);
```

Comment:
I discussed this with Menno - but the issue here is that we potentially have multiple rules for a given queue at this point, whereas we always had 1 in the previous implementation since we were iterating over rules and not queues. Since we have multiple rules, we still want to respect their `rule.meta.visibilityTimeout` value, but those values could be different, so taking the maximum of them seemed logical